### PR TITLE
mip, satp: Allow writing to these privileged registers

### PR DESCRIPTION
**Pull request by archshift**
Per the RISC-V privileged ISA, these registers should be writable.

### CODE REVIEW

File: src/register/mip.rs [hunk 1]
```diff
@@ -71,3 +71,30 @@ impl Mip {
...
+    , set_stimer, clear_stimer, 1 << 5);
+set_clear_csr!(
+    /// Machine Timer Interrupt Pending
+    , set_mtimer, clear_mtimer, 1 << 7);
+set_clear_csr!(
+    /// User External Interrupt Pending
+    , set_uext, clear_uext, 1 << 8);
+set_clear_csr!(
+    /// Supervisor External Interrupt Pending
+    , set_sext, clear_sext, 1 << 9);
```
Comment (laanwj) [inline src/register/mip.rs]: there's also a bit 11
the spec calls it MEIP
according to this naming it would be mext/"Machine External Interrupt Pending"

Comment (archshift) [reply src/register/mip.rs]: The spec says that MEIP isn’t writeable, which is why I didn’t include it here

Comment (almindor) [reply src/register/mip.rs]: Correct: ```The MEIP field in mip is a read-only bit that indicates a machine-mode external interrupt is
pending. MEIP is set and cleared by a platform-specific interrupt controller. The MEIE field in
mie enables machine external interrupts when set.```

**Comment by almindor** [comment]
@laanwj I'll let you bors this in case there was something else apart from the MEIP field.

**Pull request merged**